src/collective/elasticsearch/manager.py

```diff
@@ -0,0 +1,199 @@
+        setattr(self.catalog, CONVERTED_ATTR, True)
+        self.catalog._p_changed = True
+        mapping = getMultiAdapter((getRequest(), self), interfaces.IMappingProvider)()
+        self.connection.indices.put_mapping(body=mapping, index=self.index_name)
+
+    @property
+    def connection(self) -> Elasticsearch:
+        conn = local.get_local(self.connection_key)
+        if not conn:
+            hosts, params = utils.get_connection_settings()
+            local.set_local(self.connection_key, Elasticsearch(hosts, **params))
+            conn = local.get_local(self.connection_key)
+        return conn
+
+    def _bulk_call(self, batch):
+        data = [item for sublist in batch for item in sublist]
+        logger.info(f"Bulk call with {len(data)} entries and {len(batch)} actions.")
+        result = self.connection.bulk(index=self.index_name, body=data)
+        if "errors" in result and result["errors"] is True:
+            logger.error(f"Error in bulk operation: {result}")
+
+    def flush_indices(self):
+        self.connection.indices.flush()
+
+    def bulk(self, data: list):
+        index_name = self.index_name
+        bulk_data = []
+        for action, uuid, payload in data:
+            tmp = [
+                {action: {"_index": index_name, "_id": uuid}},
+            ]
+            if action == "index":
+                tmp.append(payload)
+            elif action == "update":
+                tmp.append({"doc": payload})
+            bulk_data.append(tmp)
+        calls = 0
+        for batch in utils.batches(bulk_data, self.bulk_size):
+            self._bulk_call(batch)
+            calls += 1
+
+    def _search(self, query, sort=None, **query_params):
+        """ """
+        if "start" in query_params:
+            query_params["from_"] = query_params.pop("start")
+
+        query_params["stored_fields"] = query_params.get("stored_fields", "path.path")
+        query_params["size"] = self.bulk_size
+
+        body = {"query": query}
+        if sort is not None:
+            body["sort"] = sort
+        warnings.simplefilter("ignore", ResourceWarning)
+        return self.connection.search(index=self.index_name, body=body, **query_params)
+
+    def search(self, query: dict, factory=None, **query_params) -> LazyMap:
+        """
+        @param query: The Plone query
+        @param factory: The factory that maps each elastic search result.
+            By default, get the plone catalog brain.
+        @param query_params: Parameters to pass to the search method
+            'stored_fields': the list of fields to get from stored source
+        """
+        factory = BrainFactory(self.catalog._catalog)
+        result = ElasticResult(self, query, **query_params)
+        return LazyMap(factory, result, result.count)
+
+    def search_results(self, request=None, check_perms=False, **kw):
+        # Make sure any pending index tasks have been processed
+        processQueue()
```

Comment:
Try removing commits from search tests, to see how they behave now.

Reply:
> I did the following steps:
> 
>     * Before installing the product, I created and published a Document with the title "Página"
> 
>     * I installed the product.
> 
>     * Converted rebuild the catalog.
> 
>     * I did a search for the word "Página".
> 

Could you retry this? @wesleybl

Reply:
Still have to try this.

Reply:
Tried to do this but some tests with fail. I will add an issue to refactor all tests, but not for this release.